zipline/src/commonMain/kotlin/app/cash/zipline/ZiplineScoped.kt

```diff
@@ -29,3 +31,51 @@ interface ZiplineScoped {
    */
   val scope: ZiplineScope
 }
+
+/**
+ * Returns a service that shares the same target object as this, but that uses [scope] to manage
+ * closing returned services.
+ *
+ * Use this function to apply different scopes to objects with different lifetimes. For example,
+ * this uses a narrower scope to close two services once they're unneeded.
+ *
+ * ```kotlin
+ * val scopeA = ZiplineScope()
+ * val quoteService = zipline.take<QuoteService>("quoteService", scopeA)
+ *
+ * val appl: LiveChart = quoteService.chart("APPL")
+ *
+ * val scopeB = ZiplineScope()
+ * val quoteServiceWithScopeB = quoteService.withScope(scopeB)
+ * val nke: LiveChart = quoteServiceWithScopeB.chart("NKE")
+ * val sq: LiveChart = quoteServiceWithScopeB.chart("SQ")
+ * scopeB.close() // closes nke and sq.
+ *
+ * scopeA.close() // closes quoteService and appl.
+ * ```
+ *
+ * This may also be used to extend the lifetime of an object beyond the lifetime of the object that
+ * produced it.
+ *
+ * ```kotlin
+ * val scopeA = ZiplineScope()
+ * val quoteService = zipline.take<QuoteService>("quoteService", scopeA)
+ *
+ * val scopeB = ZiplineScope()
+ * val quoteServiceWithScopeB = quoteService.withScope(scopeB)
+ * val nke: LiveChart = quoteServiceWithScopeB.chart("NKE")
+ *
+ * scopeA.close() // closes quoteService.
+ *
+ * nke.setPeriod(...)
+ * nke.setPrecision(...)
+ * scopeB.close() // closes nke.
+ * ```
+ *
+ * Note that closing the returned service will close this: the two handles target the same object.
+ * Closing [scope] will not close either.
```

Comment:
This seems wrong, no? Your code example above contradicts this. Did you mean something like
```suggestion
 * Closing either will not close [scope].
```
?